### ISSUE TITLE
IndentationCheck reports False-Positive while checking anonymous classes 

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#2648]
IndentationCheck reports FP while checking anonymous classes.

**Input file: `IndentationTest.java`**
```java
public class Test {
    private static String geLabel(String issue) {
        final String label =  Iterables.tryFind(list, new Predicate<String>() { // line 4
            @Override
            public boolean apply(String input) {
                return true;
            } // line 8
        }).orNull(); // line 9

        return "";
    }
}
```

**Configuration file: `my_check.xml`**
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <module name="TreeWalker">
      <module name="Indentation">
	      <property name="basicOffset" value="4"/>
	      <property name="braceAdjustment" value="0"/>
	      <property name="caseIndent" value="4"/>
    </module>
    </module>
</module>
```

**Checkstyle 6.12.1 output**
```
andreiselkin@andreiselkin ~/Downloads/temp $ java -jar checkstyle-6.12.1-all.jar -c my_check.xml IndentationTest.java
Starting audit...
/home/andreiselkin/Downloads/temp/IndentationTest.java:9: warning: '.' have incorrect indentation level 10, expected level should be 12.
Audit done.
```

1) Expected indentation level is 12 as right curly brace on line 8. Why not as start token 'final' on line 4?

2) 
> '.' have incorrect indentation level 

Why does the Check report 'dot' token, but not `}` ?

**Comment by rnveach** [comment]
6.17 reports no violations.

````
$ cat TestClass.java
public class Test {
    private static String geLabel(String issue) {
        final String label =  Iterables.tryFind(list, new Predicate<String>() { // line 4
            @Override
            public boolean apply(String input) {
                return true;
            } // line 8
        }).orNull(); // line 9

        return "";
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <module name="TreeWalker">
      <module name="Indentation">
          <property name="basicOffset" value="4"/>
          <property name="braceAdjustment" value="0"/>
          <property name="caseIndent" value="4"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-6.17-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

**Comment by romani** [comment]
@rnveach , thanks a lot!
